gtsam/linear/SparseEigen.h

```diff
@@ -0,0 +1,119 @@
+/* ----------------------------------------------------------------------------
```

Comment:
naming convention for non-class files starts with lowercase, i.e. sparseEigen.h. But, an even better suggetion is to rename SpMat below (not a good name) to SparseEigen. Then we can keep the filename and this file is about how to create one.

Reply:
done